easybuild/easyblocks/generic/bundle.py

```diff
@@ -321,8 +320,37 @@ def install_step(self):
                             new_val = path
                         env.setvar(envvar, new_val)
 
-            # close log for this component
-            comp.close_log()
+    def make_module_req_guess(self):
+        """
+        Set module requirements from all comppnents, e.g. $PATH, etc.
+        During the install step, we only set these requirements temporarily.
+        Later on when building the module, those paths are not considered.
+        Therefore, iterate through all the components again and gather
+        the requirements.
+
+        Do not remove duplicates or check for existance of folders,
+        as this is done in the generic EasyBlock while creating
+        the modulefile already.
+        """
+        # Start with the paths from the generic EasyBlock.
+        # If not added here, they might be missing entirely and fail sanity checks.
+        final_reqs = super(Bundle, self).make_module_req_guess()
+
+        for cfg, comp in self.comp_instances:
+            self.log.info("Gathering module paths for component %s v%s", cfg['name'], cfg['version'])
+            reqs = comp.make_module_req_guess()
+
+            try:
+                for key, value in sorted(reqs.items()):
+                    if isinstance(reqs, string_type):
+                        value = [value]
+                    final_reqs.setdefault(key, [])
+                    final_reqs[key] += value
+            except AttributeError:
```

Comment:
Hmm, when would we see an `AttributeError` here exactly?

If this is meant to catch the case where `make_module_req_guess` is somehow not defined for that component, then the call to `comp.make_module_req_guess` should be inside the `try`?

Reply:
The main idea is to bail out if an EasyBlock does not pass a `dict` for some reason when we gather the requiements with `comp.make_module_req_guess`, see https://github.com/easybuilders/easybuild-easyblocks/pull/3472#discussion_r1834361337

Reply:
I can add a comment to this, just so that it is clear to a future reader why we catch this exception.

Reply:
Added a comment and rebased to current `develop`.